.github/workflows/stale.yml

```diff
@@ -1,60 +1,129 @@
-# Summary: perform stale issue & PR handling on a schedule
-# For more info, see https://github.com/actions/stale/
+# Summary: perform scheduled handling of stale issues and PRs.
+# For more info, see https://github.com/actions/stale/.
 
-name: "Label and close stale issues & PRs"
+name: "Stale issues & PRs handler"
 
 on:
   schedule:
-  - cron: "0 0 * * *"
+  - cron: '0 0 * * *'
+
   workflow_dispatch:
     inputs:
       debug:
         # Note: the job body sets `debug-only` to false by default. The value
         # of inputs.debug is an empty string unless this workflow is invoked
         # manually. When it's invoked manually, GitHub's GUI presents the user
-        # with a checkbox for this flag; we default that checkbox to true
+        # with a checkbox for this flag. We default that checkbox to true
         # because the most likely reason for a manual run is debugging.
-        description: "Run in debug mode (dry run)"
+        description: 'Run in debug mode (dry run)'
         type: boolean
         default: true
 
+env:
+  # We set variables for the duration values here, so that they can be used
+  # in issue messages instead of hardcoding the values all over the place.
+  days-before-stale: 90
+  days-before-close: 60
+
+  # The syntax for actions/stale uses a comma-separated list of labels, but
+  # it's hard to read and convenient to edit in that format. We can use a
+  # normal YAML list and convert it later.
+  exempt-issue-labels-list:
+    kind/design-issue
+    kind/health
+    kind/roadmap-item
+    kind/task
+    triage/accepted
+    triage/discuss
+  exempt-pr-labels-list:
+    kind/roadmap-item
+    triage/accepted
+
+# Declare default permissions as read only.
+permissions: read-all
+
+# Cancel any previously-started but still active runs on the same branch.
+concurrency:
+  cancel-in-progress: true
+  group: ${{github.workflow}}-${{github.event.pull_request.number||github.ref}}
+
 jobs:
   stale:
     name: Label and/or close stale issues and PRs
     runs-on: ubuntu-22.04
+    timeout-minutes: 10
     permissions:
       issues: write
       pull-requests: write
     steps:
-    - uses: actions/stale@5bef64f19d7facfb25b37b414482c7164d639639 # v9
+    - name: Convert data types
+      id: converted
+      run: |
+        issues=$(echo "${{env.exempt-issue-labels-list}}" | tr ' ' ',')
+        prs=$(echo "${{env.exempt-pr-labels-list}}" | tr ' ' ',')
+        echo "exempt-issue-labels='$issues'" >> "$GITHUB_OUTPUT"
+        echo "exempt-pr-labels='$prs'" >> "$GITHUB_OUTPUT"
```

Comment:
Please use the [join()](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#join) function instead.  It can be used directly as the value of `exempt-issue-labels` below.  We can avoid writing to GITHUB_OUTPUT here.

Reply:
I originally tried that, and could not get it to work. That's why I ended up doing this indirection.